databricks/sdk/mixins/sql.py

```diff
@@ -0,0 +1,231 @@
+        :param catalog: str (optional)
+          Sets default catalog for statement execution, similar to `USE CATALOG` in SQL.
+        :param schema: str (optional)
+          Sets default schema for statement execution, similar to `USE SCHEMA` in SQL.
+        :param timeout: timedelta (optional)
+          Timeout after which the query is cancelled. If timeout is less than 50 seconds,
+          it is handled on the server side. If the timeout is greater than 50 seconds,
+          Databricks SDK for Python cancels the statement execution and throws `TimeoutError`.
+        :return: ExecuteStatementResponse
+        """
+        # The wait_timeout field must be 0 seconds (disables wait),
+        # or between 5 seconds and 50 seconds.
+        wait_timeout = None
+        if 5 <= timeout.total_seconds() <= 50:
+            # set server-side timeout
+            wait_timeout = f"{timeout.total_seconds()}s"
+
+        _LOG.debug(f"Executing SQL statement: {statement}")
+
+        # technically, we can do Disposition.EXTERNAL_LINKS, but let's push it further away.
+        # format is limited to Format.JSON_ARRAY, but other iterations may include ARROW_STREAM.
+        immediate_response = self.execute_statement(warehouse_id=warehouse_id,
+                                                    statement=statement,
+                                                    catalog=catalog,
+                                                    schema=schema,
+                                                    disposition=Disposition.INLINE,
+                                                    format=Format.JSON_ARRAY,
+                                                    byte_limit=byte_limit,
+                                                    wait_timeout=wait_timeout)
+
+        if immediate_response.status.state == StatementState.SUCCEEDED:
+            return immediate_response
+
+        self._raise_if_needed(immediate_response.status)
+
+        attempt = 1
+        status_message = "polling..."
+        deadline = time.time() + timeout.total_seconds()
+        while time.time() < deadline:
+            res = self.get_statement(immediate_response.statement_id)
+            if res.status.state == StatementState.SUCCEEDED:
+                return ExecuteStatementResponse(manifest=res.manifest,
+                                                result=res.result,
+                                                statement_id=res.statement_id,
+                                                status=res.status)
+            status_message = f"current status: {res.status.state.value}"
+            self._raise_if_needed(res.status)
+            sleep = attempt
+            if sleep > 10:
+                # sleep 10s max per attempt
+                sleep = 10
+            _LOG.debug(f"SQL statement {res.statement_id}: {status_message} (sleeping ~{sleep}s)")
+            time.sleep(sleep + random.random())
```

Comment:
Sleeping is not required as getting statement result uses server side long poll.

Reply:
@sander-goos do you mean that every request will take 10 seconds to execute? I'm not comfortable with not doing any sleeping at all, as we'll DDoS the service